olive/evaluator/olive_evaluator.py

```diff
@@ -94,13 +105,28 @@ def _evaluate_custom(
         eval_func,
         device: Device = Device.CPU,
         post_func=None,
-    ) -> Dict[str, Any]:
+    ) -> MetricResult:
         # TODO: Change the evaluate function to accept the metric rather than
         # breaking it into multiple arguments
         # return eval_func(model, metric, dataloader, device, post_func)
-        return eval_func(model, metric.user_config.data_dir, metric.user_config.batch_size, device)
+        raw_res = eval_func(model, metric.user_config.data_dir, metric.user_config.batch_size, device)
+        metric_res = {}
+        for sub_type in metric.sub_types:
+            if isinstance(raw_res, Number):
+                assert len(metric.sub_types) == 1, "Only one sub type is allowed for single value custom metric"
+                metric_res[sub_type.name] = SubMetricResult(
+                    value=raw_res, priority=sub_type.priority, higher_is_better=sub_type.higher_is_better
+                )
+            elif isinstance(raw_res, Dict):
```

Comment:
dict?

Reply:
Should be Dict from typing. `dict` is available after python3.9. 
https://stackoverflow.com/questions/37087457/difference-between-defining-typing-dict-and-dict

Reply:
Updated to dict.